### PR TITLE
Heatmap updates

### PR DESCRIPTION
**Pull request by loftusa**
- adds section in heatmap tutorial for hierarchy labels
- adds ticklabels functionality in heatmap function that seaborn has
- small bug fix and error checking changes

### CODE REVIEW

File: graspologic/plot/plot.py [hunk 1]
```diff
@@ -262,18 +263,22 @@ def heatmap(
         if len(xticklabels) != X.shape[1]:
             msg = "xticklabels must have same length {}.".format(X.shape[1])
             raise ValueError(msg)
-    elif not isinstance(xticklabels, bool):
-        msg = "xticklabels must be a bool or a list, not {}".format(type(xticklabels))
+
+    elif not isinstance(xticklabels, (bool, int)):
+        msg = "xticklabels must be a bool, int, or a list, not {}".format(
+            type(xticklabels)
+        )
```
Comment (loftusa) [inline graspologic/plot/plot.py:270]: seaborn accepts an `int` as input to this, so we should too

File: graspologic/plot/plot.py [hunk 2]
```diff
@@ -291,6 +296,11 @@ def heatmap(
         msg = "cbar must be a bool, not {}.".format(type(center))
         raise TypeError(msg)
 
+    # Warning on labels
+    if (inner_hier_labels is None) and (outer_hier_labels is not None):
+        msg = "outer_hier_labels requires inner_hier_labels to be used."
+        warnings.warn(msg)
+
```
Comment (loftusa) [inline graspologic/plot/plot.py:303]: added warning b/c as afaik `outer_hier_labels` doesn't do anything if `inner_hier_labels` isn't used

File: graspologic/plot/plot.py [hunk 3]
```diff
@@ -1267,6 +1315,8 @@ def _unique_like(vals):
 # assume that the graph has already been plotted in sorted form
 def _plot_groups(ax, graph, inner_labels, outer_labels=None, fontsize=30):
     inner_labels = np.array(inner_labels)
+    if outer_labels is not None:
+        outer_labels = np.array(outer_labels)
```
Comment (loftusa) [inline graspologic/plot/plot.py:1319]: there was some other code that did this, but I got an error when I input `outer_labels` as a list

**Comment by bdpedigo** [comment]
why make this a wrapper, as opposed to an option?

**Comment by loftusa** [comment]
> why make this a wrapper, as opposed to an option?

Thought it was a little more clunky as an option. Can make it a wrapper if you'd prefer that though

File: pytest.ini [hunk 4]
```diff
@@ -1,10 +0,0 @@
-[pytest]
```
Comment (bdpedigo) [inline pytest.ini:1]: whats up with this change?

Comment (loftusa) [reply pytest.ini]: Woops, thanks, good catch. Think I deleted that locally trying to get some VScode stuff to work. Added back in

File: graspologic/plot/plot.py [hunk 5]
```diff
@@ -341,6 +351,53 @@ def heatmap(
...
+
+    Parameters
+    ----------
+    X : nx.Graph or np.ndarray object
+        Unweighted graph or numpy matrix to plot.
+
+    colors : list-like or np.ndarray
+        A list of exactly two colors to use for the heatmap.
+
+    colorbar_ticklabels : list-like
```
Comment (bdpedigo) [inline graspologic/plot/plot.py:371]: if I don't want this, can it be turned off (like it can be for heatmap)

Comment (bdpedigo) [reply graspologic/plot/plot.py:371]: looks like yes, can you add somewhere in docs?

**Comment by bdpedigo** [comment]
@loftusa what's the status of this PR, left a few comments above but wondering if we should push in or close?

**Comment by loftusa** [comment]
> @loftusa what's the status of this PR, left a few comments above but wondering if we should push in or close?

I think probably `binary_heatmap` is unnecessary / overengineering, but the additions I made to the tutorial + the bug fixes are still useful. I'll just take that part out, keep the useful stuff, and re-request review.

File: tests/test_plot.py [hunk 6]
```diff
@@ -12,6 +12,7 @@
     heatmap,
     pairplot,
     pairplot_with_gmm,
+    binary_heatmap,
```
Comment (bdpedigo) [inline tests/test_plot.py:15]: dont need this?

Comment (loftusa) [reply tests/test_plot.py]: agreed, fixed

**Review by bdpedigo — APPROVED** [review]

**Comment by bdpedigo** [comment]
@loftusa any interest in getting this merged? if not will close for the time being

**Comment by loftusa** [comment]
> @loftusa any interest in getting this merged? if not will close for the time being

Yeah - the doc-building error was weird and I wasn't sure how to fix, then other stuff took priority...

```
TypeError: 'generator' object is not reversible

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/graspologic/envs/750/lib/python3.8/site-packages/sphinx/util/nodes.py", line 94, in apply_source_workaround
    for classifier in reversed(node.parent.traverse(nodes.classifier)):
TypeError: 'generator' object is not reversible
```

**Comment by bdpedigo** [comment]
I dont see any current doc errors, just a format check?

**Comment by bdpedigo** [comment]
> I dont see any current doc errors, just a format check?

bump @loftusa

**Comment by bdpedigo** [comment]
oops didnt mean to hit close

**Comment by loftusa** [comment]
`AttributeError: type object 'gensim._matutils.array' has no attribute '__reduce_cython__' `

what the heck is this

**Comment by bdpedigo** [comment]
seeing if the issue fixed itself 🤞

**Pull request merged**